src/plugin/sdl/sdl.c

```diff
@@ -225,7 +191,88 @@ int SDL_priv_init(void)
   return 0;
 }
 
-int SDL_init(void)
+static void SDL_text_init(void)
+#if defined(HAVE_SDL2_TTF) && defined(HAVE_FONTCONFIG)
+{
+  int err;
+  char *pth = NULL;
+  FcPattern *pat, *match;
+  FcResult result;
+  char *foundname;
+
+  if (!config.sdl_font[0])
+    goto tidy_bitmap;
+
+  err = TTF_Init();
+  if (err) {
+    error("TTF_Init: %s\n", TTF_GetError());
+    goto tidy_bitmap;
+  }
+
+  // Lookup font path with fontconfig
+  if (!FcInit()) {
+    error("FcInit: returned false\n");
+    goto tidy_ttf;
+  }
+  pat = FcNameParse((const FcChar8*)config.sdl_font);
+  if (!pat)
+    goto tidy_ttf;
+  FcConfigSubstitute(NULL, pat, FcMatchPattern);
+  FcDefaultSubstitute(pat);
+  match = FcFontMatch(NULL, pat, &result);
+  if (!match) {
+    FcPatternDestroy(pat);
+    goto tidy_ttf;
+  }
+  FcPatternGetString(match, FC_FAMILY, 0, (FcChar8 **)&foundname);
+  FcPatternGetString(match, FC_FILE, 0, (FcChar8 **)&pth);
+  FcPatternGetInteger(match, FC_INDEX, 0, &sdl_font_idx);
+
+  // Fontconfig guesses if not an exact match, which is not what we want
+  if (strcasecmp(config.sdl_font, foundname) != 0) {
+    v_printf("SDL: not accepting substitute font '%s'\n", foundname);
+    FcPatternDestroy(match);
+    FcPatternDestroy(pat);
+    goto tidy_ttf;
+  }
+  v_printf("SDL: using font '%s(%d)'\n", pth, sdl_font_idx);
+
+  sdl_font_rw = SDL_RWFromFile(pth, "r");
+  FcPatternDestroy(match);
+  FcPatternDestroy(pat);
+  if (!sdl_font_rw) {
+    error("SDL_RWFromFile: %s\n", SDL_GetError());
+    goto tidy_ttf;
+  }
+
+  if (config.X_winsize_x >= MIN_X && config.X_winsize_y >= MIN_X) {
+    if (!setup_ttf_winsize(config.X_winsize_x, config.X_winsize_y))
+      goto tidy_font_rw;
+  } else {
+    if (!setup_ttf_pntsize(22))
+      goto tidy_font_rw;
+  }
+
+  register_text_system(&Text_SDL);
+  use_bitmap_font = 0;
```

Comment:
But its already 0?

Reply:
This function is the equivalent of `init_x11_support()` which I removed. `use_bitmap_font` is not set anywhere else, should it be?

Reply:
Its just set in bss (to 0) but as you prefer.